lib/ecmascript.ts

```diff
@@ -1595,7 +1853,14 @@ export function CalendarMergeFields(calendar, fields, additionalFields) {
   return result;
 }
 
-export function CalendarDateAdd(calendar, date, duration, options, dateAdd?: any) {
+export function CalendarDateAdd(
```

Comment:
Nit: Mind adding a comment here + elsewhere that using the `Parameters` pattern here for typing arguments is especially important given that user-defined calendars are a thing and we (presumably) partially document our expectations within the .d.ts.

Reply:
These are getting cleaned up in a later commit. The new code will look like this: 
```ts
export function CalendarDateAdd(
  calendar: Temporal.CalendarProtocol,
  date: CalendarProtocolParams['dateAdd'][0],
  duration: CalendarProtocolParams['dateAdd'][1],
  options: CalendarProtocolParams['dateAdd'][2],
  dateAddParam?: Temporal.CalendarProtocol['dateAdd']
) {
```

Better?

I don't think it makes sense to add a comment everywhere these are used because we'd end up with 100s of comments. Seems better to just document the *Params types in one place. Let's discuss when we review the later commit where this changes.